### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by VascoVerissimo**

### CODE REVIEW

File: test/io.jl [hunk 1]
```diff
@@ -11,22 +11,40 @@ extend this in the original package
...
-cwd = pwd()
+#create genData and data folder and change dir to dataPath
+if occursin("jenkins", homedir()) || "TRAVIS" in keys(ENV)
+    cwd = pwd()
+    genDataPath = mktempdir()
+    dataPath = mktempdir()
+    cd(dataPath)
+else
+    # cd("test")
+    cwd = pwd()
```
Comment (laurentheirendt) [inline test/io.jl]: Here, `pwd()` must always be `/test`.

File: test/io.jl [hunk 2]
```diff
@@ -11,22 +11,40 @@ extend this in the original package
...
+    if !isdir("genData")
```
Comment (laurentheirendt) [inline test/io.jl]: This can be written in a loop & array.

File: test/io.jl [hunk 3]
```diff
@@ -11,22 +11,40 @@ extend this in the original package
...
-download("http://imlspenticton.uzh.ch/robinson_lab/cytofWorkflow/PBMC8_fcs_files.zip", "PBMC8_fcs_files.zip")
-run(`unzip PBMC8_fcs_files.zip`)
+# fetch the required data for testing and download the zip archive and unzip it
+dataFiles = ["PBMC8_metadata.xlsx", "PBMC8_panel.xlsx", "PBMC8_fcs_files.zip"]
+for f in dataFiles
+    if !isfile(f)
+        download("http://imlspenticton.uzh.ch/robinson_lab/cytofWorkflow/"*f, f)
+        if occursin(".zip", f)
+            run(`unzip PBMC8_fcs_files.zip`)
+        else
```
Comment (laurentheirendt) [inline test/io.jl]: There is no need for `else` here.

File: test/refData/ref_parallel_df_codes.csv [hunk 4]
```diff
@@ -0,0 +1,101 @@
+CD3(110:114)Dd,CD45(In115)Dd,CD4(Nd145)Dd,CD20(Sm147)Dd,CD33(Nd148)Dd,CD123(Eu151)Dd,CD14(Gd160)Dd,IgM(Yb171)Dd,HLA_DR(Yb174)Dd,CD7(Yb176)Dd
```
Comment (laurentheirendt) [inline test/refData/ref_parallel_df_codes.csv:1]: Would it be possible to use a single ref_data set?

**Review by laurentheirendt — CHANGES_REQUESTED** [review]

File: test/io.jl (hunk 1 above)
Comment (VascoVerissimo) [reply test/io.jl]: done

File: test/io.jl (hunk 3 above)
Comment (VascoVerissimo) [reply test/io.jl]: done

File: test/refData/ref_parallel_df_codes.csv (hunk 4 above)
Comment (VascoVerissimo) [reply test/refData/ref_parallel_df_codes.csv:1]: I dont think so, the batch and the parallel methods generate different `df_codes` files

File: test/io.jl (hunk 2 above)
Comment (VascoVerissimo) [reply test/io.jl]: done

File: test/io.jl [hunk 5]
```diff
@@ -11,22 +11,39 @@ extend this in the original package
...
+#create genData and data folder and change dir to dataPath
 cwd = pwd()
+if occursin("jenkins", homedir()) || "TRAVIS" in keys(ENV)
+    genDataPath = mktempdir()
+    dataPath = mktempdir()
+    cd(dataPath)
+else
+    if !occursin("test", cwd)
+        cd("test")
+        cwd = pwd()
```
Comment (laurentheirendt) [inline test/io.jl:19]: This line is not required.

File: test/io.jl [hunk 6]
```diff
@@ -11,22 +11,39 @@ extend this in the original package
...
+    end
+    dataFolders = ["genData", "data"]
+    for i in dataFolders
+        if !isdir(i)
+            mkdir(i)
```
Comment (laurentheirendt) [inline test/io.jl]: `i` is generally reserved for counters. May you use `dir` or `directory`?

File: test/io.jl [hunk 7]
```diff
@@ -11,22 +11,39 @@ extend this in the original package
...
+        end
+    end
+    genDataPath = cwd*"/genData"
+    dataPath = cwd*"/data"
+    cd(dataPath)
```
Comment (laurentheirendt) [inline test/io.jl]: This line is redundant with line 19 and can be placed outside the `if` condition.

File: test/runtests.jl [hunk 8]
```diff
@@ -1,51 +1,93 @@
...
+using Distributed
 
 #fix the seed
 Random.seed!(1)
 
 include("io.jl")
 
-#BATCH & PARALLEL
+#test cleannames
+@testset "cleannames" begin
```
Comment (laurentheirendt) [inline test/runtests.jl]: May you put these tests into external file(s)?

**Review by laurentheirendt — CHANGES_REQUESTED** [review]

File: test/runtests.jl (hunk 8 above)
Comment (VascoVerissimo) [reply test/runtests.jl]: ok, like the `parallel.jl`?

File: test/io.jl (hunk 6 above)
Comment (VascoVerissimo) [reply test/io.jl]: changed

File: test/io.jl (hunk 7 above)
Comment (VascoVerissimo) [reply test/io.jl]: done

File: test/runtests.jl (hunk 8 above)
Comment (VascoVerissimo) [reply test/runtests.jl]: done

File: test/io.jl (hunk 5 above)
Comment (laurentheirendt) [reply test/io.jl:19]: Actually, no need 👍

**Review by laurentheirendt — APPROVED** [review]

**Comment by laurentheirendt** [comment]
Looks good! Well done @VascoVerissimo 👍

**Pull request merged**